lien de l'image docker utilisée : 
https://github.com/rnakato/ShortCake

d'abord lancer : 
```{bash}
sbatch "7_Article/script/script_Val/0_launch_jupyter.sh"
```

In [2]:
library (Seurat)

Loading required package: SeuratObject

Loading required package: sp


Attaching package: ‘SeuratObject’


The following objects are masked from ‘package:base’:

    intersect, t




In [ ]:
# load the Seurat object
path_RNA_combined  <- "7_Article/results/tab_seurat_correct/RNA_combined.rds"
RNA_combined <- readRDS(path_RNA_combined)

path_RNA_rep1  <- "7_Article/results/tab_seurat_correct/RNA_rep1.rds"
RNA_rep1 <- readRDS(path_RNA_rep1)

path_RNA_rep2  <- "7_Article/results/tab_seurat_correct/RNA_rep2.rds"
RNA_rep2 <- readRDS(path_RNA_rep2)


In [9]:
str (RNA_combined)

Formal class 'Seurat' [package "SeuratObject"] with 13 slots
  ..@ assays      :List of 1
  .. ..$ RNA:Formal class 'Assay5' [package "SeuratObject"] with 8 slots
  .. .. .. ..@ layers    :List of 1
  .. .. .. .. ..$ counts:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. .. ..@ i       : int [1:11781869] 617 845 848 1346 2735 2819 2825 2924 3200 4020 ...
  .. .. .. .. .. .. ..@ p       : int [1:430176] 0 0 12 12 12 64 72 73 132 132 ...
  .. .. .. .. .. .. ..@ Dim     : int [1:2] 4422 430175
  .. .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. ..@ x       : num [1:11781869] 0.143 1 0.143 0.143 0.143 ...
  .. .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ cells     :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. .. .. .. ..@ .Data: logi [1:430175, 1] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. .. .. ..

Test avec le dataset fusionné 

🔹 1. 4422 features → Nombre de gènes détectés
	•	Un “feature” dans Seurat correspond généralement à un gène (mais cela peut inclure des transcrits non codants).
	•	4422 signifie que l’objet contient 4 422 gènes différents.
	•	Ce nombre dépend du prétraitement, des filtres appliqués et du type de données (ex. si les gènes ribosomiques ont été supprimés).

⸻

🔹 2. 430175 samples → Nombre de cellules analysées
	•	En scRNA-seq, chaque “sample” correspond à une cellule unique.
	•	430 175 cellules → Cela indique une très grande expérience, potentiellement issue d’un séquençage très profond ou de plusieurs échantillons combinés.

⚠️ Ce nombre semble extrêmement élevé pour du single-cell RNA-seq.


In [7]:
RNA_combined

An object of class Seurat 
4422 features across 430175 samples within 1 assay 
Active assay: RNA (4422 features, 0 variable features)
 1 layer present: counts

In [ ]:
#Filtering cells based on number of genes and number of UMIs
#RNA_combined <- subset(RNA_combined, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & nCount_RNA > 800 & nCount_RNA < 20000)

#Filtering genes based on number of cells
#RNA_combined <- subset(RNA_combined, subset = nFeature_RNA > 3)

#Filtering cells based on number of genes and number of UMIs
RNA_combined <- subset(RNA_combined, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & nCount_RNA > 800 & nCount_RNA < 20000)



ERROR: Error in parse(text = input): <text>:4:5: unexpected symbol
3: 
4: min nFeature_RNA
       ^


In [5]:
RNA_combined

An object of class Seurat 
4422 features across 3444 samples within 1 assay 
Active assay: RNA (4422 features, 0 variable features)
 1 layer present: counts

In [4]:
RNA_combined <- NormalizeData(object = RNA_combined)

Normalizing layer: counts



In [5]:
RNA_combined <- FindVariableFeatures(object = RNA_combined)

Finding variable features for layer counts



In [6]:
RNA_combined <- ScaleData(object = RNA_combined)

Centering and scaling data matrix



In [7]:
RNA_combined <- RunPCA(object = RNA_combined)

: 

: 

: 

In [ ]:
RNA_combined <- FindNeighbors(object = RNA_combined, dims = 1:30)

In [ ]:
RNA_combined <- FindClusters(object = RNA_combined)

In [ ]:
RNA_combined <- RunUMAP(object = RNA_combined, dims = 1:30)

In [ ]:
DimPlot(object = RNA_combined, reduction = "umap")